src/slick.grid.ts

```diff
@@ -5931,29 +5933,35 @@ export class SlickGrid<TData = any, C extends Column<TData> = Column<TData>, O e
     this.getEditorLock()?.activate(this.editController as EditController);
     this.activeCellNode.classList.add('editable');
 
-    const useEditor: any = editor || this.getEditor(this.activeRow, this.activeCell);
+    const useEditor = editor || this.getEditor(this.activeRow, this.activeCell);
 
     // don't clear the cell if a custom editor is passed through
-    if (!editor && !useEditor.suppressClearOnEdit) {
+    if (!editor && !useEditor?.suppressClearOnEdit) {
       Utils.emptyElement(this.activeCellNode);
     }
 
     let metadata = (this.data as CustomDataView<TData>)?.getItemMetadata?.(this.activeRow);
     metadata = metadata?.columns as any;
     const columnMetaData = metadata && (metadata[columnDef.id as keyof ItemMetadata] || (metadata as any)[this.activeCell]);
 
-    this.currentEditor = new useEditor({
+    //the editor must be constructable
+    if (typeof useEditor !== 'function') {
+      return;
+    }
+    const editorArgs: EditorArguments<TData, C, O> = {
       grid: this,
       gridPosition: this.absBox(this._container),
       position: this.absBox(this.activeCellNode),
       container: this.activeCellNode,
       column: columnDef,
       columnMetaData,
       item: item || {},
-      event: e,
+      event: e as Event,
```

Comment:
Typescript requires narrowing down to one type, because EditorArguments.event is only of type Event.
It wont make a diffenece in the generated output.